testbed/tests/window/test_window.py

```diff
@@ -507,6 +725,167 @@ async def test_move_and_resize(second_window, second_window_probe):
         assert second_window.size == (250 + extra_width, 210 + extra_height)
         assert second_window_probe.content_size == (250, 210)
 
+    @pytest.fixture(scope="session")
+    def intermediate_states_cycle():
+        # Use a iterator cycling fixture to ensure each state is the initial
+        # intermediate state at least once and to test specific problematic
+        # combinations. This is to ensure full code coverage for all backends.
+        intermediate_states = [
```

Comment:
I'm afraid I don't understand what is going on here. It seems like a very complex and stateful process for determining initial test state, which is something that *should* be exceptionally boring, predictable, and obvious on first inspection.

tl;dr - resist the temptation to be "clever" when writing tests, because every time you're clever, you open yourself up for the possibility that the test isn't actually testing what you think it does.

Reply:
I have made intermediate_states into a simple pytest paramterization. I have also removed cases of intermediate states which didn't test anything useful and have kept the cases which cause errors when the implementation is incorrect.
However, due to parameterization of intermediate_states, the number of test cases has gone up, i.e., 25 x. 2 x 1 = 50 test cases. This makes the test run for longer.